web/app.js

```diff
@@ -1329,12 +1329,33 @@ const PDFViewerApplication = {
       pdfViewer.optionalContentConfigPromise.then(optionalContentConfig => {
         this.pdfLayerViewer.render({ optionalContentConfig, pdfDocument });
       });
+      if ("requestIdleCallback" in window) {
+        window.requestIdleCallback(() => {
+          this._collectTelemetry();
+        });
```

Comment:
Also, I think that you need to track this callback and ensure that it's cancelled when/if the document is closed, to avoid the possibility of more than one of these running at any one time.

Basically something like this (assuming that you add an `_idleCallbacks: new Set(),` after line https://github.com/mozilla/pdf.js/blob/master/web/app.js#L249):
```suggestion
        const callback = window.requestIdleCallback(() => {
          this._collectTelemetry(pdfDocument);
          
          this._idleCallback.delete(callback);
        });
        this._idleCallbacks.add(callback);
```

Then, e.g. after line https://github.com/mozilla/pdf.js/blob/master/web/app.js#L745 add something like
```javascript
    for (const callback of this._idleCallbacks) {
      window.cancelIdleCallback(callback);
    }
    this._idleCallbacks.clear();
```